datafusion/common/src/scalar.rs

```diff
@@ -919,12 +1056,14 @@ fn with_timezone_to_naive_datetime(
 /// This function creates the [`NaiveDateTime`] object corresponding to the
 /// given timestamp, whose tick size is specified by `UNIT_NANOS`.
 #[inline]
-fn ticks_to_naive_datetime<const UNIT_NANOS: i64>(ticks: i64) -> Result<NaiveDateTime> {
-    NaiveDateTime::from_timestamp_opt(
-        (ticks * UNIT_NANOS) / 1_000_000_000,
-        ((ticks * UNIT_NANOS) % 1_000_000_000) as u32,
-    )
-    .ok_or_else(|| {
+fn ticks_to_naive_datetime<const UNIT_NANOS: i128>(ticks: i64) -> Result<NaiveDateTime> {
```

Comment:
https://docs.rs/arrow-array/latest/arrow_array/temporal_conversions/fn.as_datetime.html

Reply:
If one of the timestamps has an incompatible timezone, or has not timezone data, they are both treated as they don't have a timezone data. Is it an acceptable behavior?

Reply:
I think if we can't parse the timezone we should return an error, silently continuing I think would potentially be quite confusing